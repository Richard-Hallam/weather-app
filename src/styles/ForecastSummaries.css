.forecast-summaries {
    display: flex;
    justify-content: space-between;
    align-items: center;
    text-align: center;
    background-color: gray;
}