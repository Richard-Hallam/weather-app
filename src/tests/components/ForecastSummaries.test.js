import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
    const validProps = [
        {
        date: new Date(111111).toDateString(),
        description:"Stub description 1",
        icon: "800",
        temperature: {
            max: 22,
            min: 12,
        },
      },
      {
        date: new Date(222222).toDateString(),
        description: "Stub description 2",
        icon: "602",
        temperature:{
            max: 24,
            min: 13,
        },
      },
    ];

    it("Renders correctly", () => {
        const { asFragment } = render(<ForecastSummaries forecasts={validProps}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("Renders the correct number of ForecastSummary instances", () => {
        const{ getAllByTestId } = render(
            <ForecastSummaries forecasts={validProps} />
        );
        expect(getAllByTestId("forecast-summary")).toHaveLength(2);
    });
});