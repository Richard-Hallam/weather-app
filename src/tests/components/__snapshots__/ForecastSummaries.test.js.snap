// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries Renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <img
          alt="Weather Icon"
          src="https://openweathermap.org/img/wn/01d@2x.png"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        22
      </div>
      <div
        class="forecast-summary_description"
      >
        Stub description 1
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <img
          alt="Weather Icon"
          src="https://openweathermap.org/img/wn/13n@2x.png"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        24
      </div>
      <div
        class="forecast-summary_description"
      >
        Stub description 2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
